vertical-pod-autoscaler/pkg/apis/poc.autoscaling.k8s.io/v1alpha1/types.go

```diff
@@ -207,3 +207,64 @@ type RecommendedContainerResources struct {
 	// +optional
 	MaxRecommended apiv1.ResourceList `json:"maxRecommended,omitempty" protobuf:"bytes,3,rep,name=maxRecommended,casttype=ResourceList,castkey=ResourceName"`
 }
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VerticalPodAutoscalerSnapshot is the snapshot of the internal state of VPA that
+// is used for recovery after recommender's restart.
+type VerticalPodAutoscalerSnapshot struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Specification of the snapshot.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status.
+	// +optional
+	Spec VerticalPodAutoscalerSnapshotSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+
+	// Data of the snapshot.
+	// +optional
+	Status VerticalPodAutoscalerSnapshotStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VerticalPodAutoscalerSnapshotList is a list of VerticalPodAutoscalerSnapshot objects.
+type VerticalPodAutoscalerSnapshotList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []VerticalPodAutoscalerSnapshot `json:"items"`
+}
+
+// VerticalPodAutoscalerSnapshotSpec is the specification of the snapshot object.
+type VerticalPodAutoscalerSnapshotSpec struct {
+	// Name of the VPA object that stored VerticalPodAutoscalerSnapshot object.
+	VPAObjectName string `json:"vpaObjectName,omitempty" protobuf:"bytes,1,opt,name=vpaObjectName"`
+}
+
+// VerticalPodAutoscalerSnapshotStatus contains data of the snapshot.
+type VerticalPodAutoscalerSnapshotStatus struct {
+	// The time when the status was last refreshed.
+	LastUpdateTime metav1.Time `json:"lastUpdateTime,omitempty" protobuf:"bytes,1,opt,name=lastUpdateTime"`
+
+	// Schema of the snapshot that identifies format of the stored data.
+	Schema string `json:"schema,omitempty" protobuf:"bytes,2,opt,name=schema"`
+
+	// Snapshots of histograms describing consumption of resources.
+	Histograms []HistogramSnapshot `json:"" protobuf:"bytes,3,rep,name=histograms"`
+}
+
+// HistogramSnapshot contains data needed to reconstruct the histogram.
+type HistogramSnapshot struct {
+	// Name of the resource that this HistogramSnapshot refers to.
+	Resource string `json:"resource,omitempty" protobuf:"bytes,1,opt,name=resource"`
```

Comment:
You should be able to remove this field if you follow the previous comment.

Reply:
See other comment.